aws-rds-dbinstance/src/main/java/software/amazon/rds/dbinstance/util/ResourceModelHelper.java

```diff
@@ -31,13 +32,20 @@ public static boolean shouldUpdateAfterCreate(final ResourceModel model) {
                                 StringUtils.hasValue(model.getMonitoringRoleArn()) ||
                                 Optional.ofNullable(model.getBackupRetentionPeriod()).orElse(0) > 0 ||
                                 Optional.ofNullable(model.getMonitoringInterval()).orElse(0) > 0 ||
-                                isStorageParametersModified(model) ||
+                                (isSqlServer(model) && isStorageParametersModified(model)) ||
                                 BooleanUtils.isTrue(model.getManageMasterUserPassword()) ||
                                 BooleanUtils.isTrue(model.getDeletionProtection()) ||
                                 BooleanUtils.isTrue(model.getEnablePerformanceInsights())
                 );
     }
 
+    public static boolean isSqlServer(final ResourceModel model) {
+        final String engine = model.getEngine();
+        // treat unknown engines as SQLServer
```

Comment:
I believe this benefits from further explanation or guidance as this might not be the case in the future when new engines get released.

Reply:
Treating unknown engines is left as a defensive programming. Unless we somehow lost desired model changes, there should not be any situation when we do not know the engine - it is fetched long before this method is accessed. Future engines are unknown at this point in time, I think we should consider them after they are known.